### PR TITLE
Support for Sim(3) alignment with missing poses

### PR DESCRIPTION
**Pull request by johnwlambert**
After the MuitlviewOptimizer, we align the poses that come out of Shonan and 1dsfm to the ground truth, and measure the pose errors.

1dsfm may return None for some poses (if the Value is missing). Previous Sim(3) alignment required a valid pose for every frame, in GT and estimate.

### CODE REVIEW

File: gtsfm/utils/geometry_comparisons.py [hunk 1]
```diff
@@ -40,6 +40,52 @@ def align_rotations(aRi_list: List[Rot3], bRi_list: List[Rot3]) -> List[Rot3]:
...
+    valid_bTi_list = []
+    for i, bTi in enumerate(bTi_list):
+        if bTi is None:
+            dropped_camera_idxs.append(i)
+        else:
+            valid_camera_idxs.append(i)
+            valid_bTi_list.append(bTi)
+            corresponding_aTi_list.append(aTi_list[i])
+
+    if len(dropped_camera_idxs) > 0:
```
Comment (akshay-krishnan) [inline gtsfm/utils/geometry_comparisons.py:75]: I think this should be logged in whichever module is dropping these (if necessary), unnecessary here.

**Review by akshay-krishnan — CHANGES_REQUESTED** [review]
I don’t think its a good idea to have a _wrapper function for this. I think we should add this code to the align_poses_sim3() itself. Is there a problem with doing that? 

Also what does Similarity3 do if it gets a None?

**Comment by johnwlambert** [comment]
Thanks for the review @akshay-krishnan. So my concern was that align_poses_sim3() is already quite long, and this would make it less readable. I don't think this code is relevant to alignment, its more just a wrapper around it

File: gtsfm/utils/geometry_comparisons.py (hunk 1 above)
Comment (ayushbaid) [reply gtsfm/utils/geometry_comparisons.py:75]: I agree that logging here is not very useful. But I don't think we need to log it anywhere. We dont care about the exact indices anyway. We can just log (and I think we are already doing it) the number of valid cameras.

**Review by ayushbaid — APPROVED** [review]
LGTM!

File: gtsfm/utils/geometry_comparisons.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/utils/geometry_comparisons.py:75]: sure, i'll remove it

**Comment by akshay-krishnan** [comment]
> Thanks for the review @akshay-krishnan. So my concern was that align_poses_sim3() is already quite long, and this would make it less readable. I don't think this code is relevant to alignment, its more just a wrapper around it

Okay in that case, can we give it a better name? Wrapper doesn’t say anything. Maybe something that includes “missing poses”? align_poses_sim3_ignore_missing()? I don’t feel strongly about it, so feel free to leave as is.

**Review by akshay-krishnan — APPROVED** [review]

**Comment by johnwlambert** [comment]
> > Thanks for the review @akshay-krishnan. So my concern was that align_poses_sim3() is already quite long, and this would make it less readable. I don't think this code is relevant to alignment, its more just a wrapper around it
> 
> Okay in that case, can we give it a better name? Wrapper doesn’t say anything. Maybe something that includes “missing poses”? align_poses_sim3_ignore_missing()? I don’t feel strongly about it, so feel free to leave as is.

Good point -- the name can be better. I like `align_poses_sim3_ignore_missing()`

**Comment by johnwlambert** [comment]
I updated the name

**Pull request merged**